ush/rocoto/setup_expt.py

```diff
@@ -79,218 +112,162 @@ def edit_baseconfig():
+        link initial condition files from $ICSDIR to $COMROT
+        """
+
+    parser = ArgumentParser(description=description,
+                            formatter_class=ArgumentDefaultsHelpFormatter)
+
+    # Set up sub-parsers for various modes of experimentation
+    subparser = parser.add_subparsers(dest='mode')
+    cycled = subparser.add_parser(
+        'cycled', help='arguments for cycled mode')
+    forecasts = subparser.add_parser(
+        'forecast-only', help='arguments for forecast-only mode')
+
+    # Common arguments across all modes
+    for subp in [cycled, forecasts]:
+        subp.add_argument('--pslot', help='parallel experiment name',
+                          type=str, required=False, default='test')
+        subp.add_argument('--resdet', help='resolution of the deterministic model forecast',
+                          type=int, required=False, default=384)
+        subp.add_argument('--comrot', help='full path to COMROT',
+                          type=str, required=False, default=os.getenv('HOME'))
+        subp.add_argument('--expdir', help='full path to EXPDIR',
+                          type=str, required=False, default=os.getenv('HOME'))
+        subp.add_argument('--idate', help='starting date of experiment, initial conditions must exist!', required=True, type=lambda dd: datetime.strptime(dd, '%Y%m%d%H'))
+        subp.add_argument('--edate', help='end date experiment', required=True, type=lambda dd: datetime.strptime(dd, '%Y%m%d%H'))
+        subp.add_argument('--icsdir', help='full path to initial condition directory', type=str, required=False, default=os.getenv('HOME'))
+        subp.add_argument('--configdir', help='full path to directory containing the config files',
+                          type=str, required=False, default=os.path.join(top,'parm/config'))
+        subp.add_argument('--cdump', help='CDUMP to start the experiment',
+                          type=str, required=False, default='gdas')
+        subp.add_argument('--gfs_cyc', help='GFS cycles to run', type=int,
+                          choices=[0, 1, 2, 4], default=1, required=False)
+        subp.add_argument('--start', help='restart mode: warm or cold', type=str,
+                          choices=['warm', 'cold'], required=False, default='cold')
+
+    # cycled mode additional arguments
+    cycled.add_argument('--resens', help='resolution of the ensemble model forecast',
+                        type=int, required=False, default=192)
+    cycled.add_argument('--nens', help='number of ensemble members',
+                        type=int, required=False, default=20)
+    cycled.add_argument('--app', help='UFS application', type=str,
+                        choices=['ATM', 'ATMW'], required=False, default='ATM')
+
+    # forecast only mode additional arguments
+    forecasts.add_argument('--app', help='UFS application', type=str, choices=[
+        'ATM', 'ATMW', 'S2S', 'S2SW'], required=False, default='ATM')
 
```

Comment:
Wondering why the two --app arguments (cycled and forecasts) can't be combined into one? I see a different `choices` for both but they overlap. Is this a way of not allowing the `S2S` and `S2SW` apps in cycled mode? It's two places to maintain that list now.